nautobot/circuits/graphql/types.py

```diff
@@ -11,3 +12,46 @@ class CircuitTerminationType(gql_optimizer.OptimizedDjangoObjectType, PathEndpoi
     class Meta:
         model = CircuitTermination
         filterset_class = CircuitTerminationFilterSet
+
+    cable_peer_circuit_termination = Field("nautobot.circuits.graphql.types.CircuitTerminationType")
+    cable_peer_front_port = Field("nautobot.dcim.graphql.types.FrontPortType")
+    cable_peer_interface = Field("nautobot.dcim.graphql.types.InterfaceType")
+    cable_peer_rear_port = Field("nautobot.dcim.graphql.types.RearPortType")
+    connected_circuit_termination = Field("nautobot.circuits.graphql.types.CircuitTerminationType")
+    connected_interface = Field("nautobot.dcim.graphql.types.InterfaceType")
+
+    def resolve_cable_peer_circuit_termination(self, args):
+        peer = self.get_cable_peer()
+        if type(peer).__name__ == "CircuitTermination":  # type built-in used to avoid class loading
+            return peer
+        return None
```

Comment:
It might be possible or desirable to write and use a generator function or two to reduce the boilerplate, something like:

```python
def cable_peer_resolver(type_name):
    def resolve_cable_peer(self, args):
        peer = self.get_cable_peer()
        if type(peer).__name__ == type_name:
            return peer
        return None
    return resolve_cable_peer

resolve_cable_peer_circuit_termination = cable_peer_resolver("CircuitTermination")
resolve_cable_peer_front_port = cable_peer_resolver("FrontPort")
resolve_cable_peer_interface = cable_peer_resolver("Interface")
...
```

Reply:
That worked brilliantly. I've refactored the code.